ext/date/php_date.stub.php

```diff
@@ -114,214 +114,211 @@ function date_sunset(
 
 function date_sun_info(int $timestamp, float $latitude, float $longitude): array {}
 
-// NB: Adding return types to methods is a BC break!
-// For now only using @return annotations here.
-
 interface DateTimeInterface
 {
-    /** @return string */
-    public function format(string $format);
+    /** @tentative-return-type */
+    public function format(string $format): string;
 
-    /** @return DateTimeZone|false */
-    public function getTimezone();
+    /** @tentative-return-type */
+    public function getTimezone(): DateTimeZone|false;
 
-    /** @return int */
-    public function getOffset();
+    /** @tentative-return-type */
+    public function getOffset(): int;
 
-    /** @return int|false */
-    public function getTimestamp();
+    /** @tentative-return-type */
+    public function getTimestamp(): int|false;
 
-    /** @return DateInterval|false */
-    public function diff(DateTimeInterface $targetObject, bool $absolute = false);
+    /** @tentative-return-type */
+    public function diff(DateTimeInterface $targetObject, bool $absolute = false): DateInterval;
```

Comment:
I found that the `false` type is superfluous.

P.S. @nikic I think your idea will turn out to be brilliant :) If we have real return type declarations (+ `@tentative-return-type`), PHPStorm will complain for the stubs if there is any incompatibility issue. So this addresses my concern which I raised in https://github.com/php/php-src/pull/6548#issuecomment-835397737